### PR TITLE
languages: add MOOS guide

### PR DESCRIPTION
**Pull request by kyrofa**
My take of the user-story-driven docs for MOOS developers.

### CODE REVIEW

File: build-snaps/moos.md [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+Here are some snap advantages that will benefit many MOOS projects:
+
+ * Bundle all the runtime requirements, including the exact version of MOOS/MOOS-IvP, system libraries, etc.
+ * Simplify installation instructions, regardless of distribution, to `snap install mymoosapp`.
+ * Directly control the delivery of application updates.
+ * Extremely simple creation of daemons.
+
+## Getting started
+
+Let's take a look at ping/pong MOOS example application, and show how simple it is to snap.
```
Comment (evandandrea) [inline build-snaps/moos.md]: A ping pong

File: build-snaps/moos.md [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+
+```
+snapcraft help cmake
+```
+
+### Extending and overriding behaviour
+
+You can [extend the behaviour](/build-snaps/scriptlets) of any part in your `snapcraft.yaml` with shell commands. These can be run after pulling the source code but before building by using the `prepare` keyword. The build process can be overridden entirely using the `build` keyword and shell commands. The `install` keyword is used to run shell commands after building your code, useful for making post build modifications such as relocating build assets.
+
+For example, let's pretend that the MOOS examples contained a suite of tests (they don't, so this won't actually work). We could run that test suite at the end of the build. If these tests failed, the snap creation would be terminated:
```
Comment (evandandrea) [inline build-snaps/moos.md]: Is there really no functional example we can give for scrollers? It’s weird to say “don’t try this, it won’t work.”

Comment (kyrofa) [reply build-snaps/moos.md]: I'm afraid not. This is the best pre-baked example we have, and it doesn't even have install rules much less tests :cry: . We can remove this section entirely, of course.

Comment (evandandrea) [reply build-snaps/moos.md]: Are MOOS developers likely to need scriplets? If not, I’d drop it. If they are, well, it doesn’t have to be tests.

File: build-snaps/moos.md (hunk 1 above)
Comment (evandandrea) [reply build-snaps/moos.md]: Or the

File: build-snaps/moos.md (hunk 2 above)
Comment (kyrofa) [reply build-snaps/moos.md]: Not necessarily. I'll go ahead and drop this section, at least for now. We can add it back if we come up with a brilliant idea.

Comment (kyrofa) [reply build-snaps/moos.md]: Done.

File: build-snaps/moos.md (hunk 1 above)
Comment (kyrofa) [reply build-snaps/moos.md]: Done. Went with "the".

**Comment by evandandrea** [comment]
We’ll need a PR to link to it, but let’s get this landed 👍

**Comment by sergiusens** [comment]
well... that was quick ;-)

should I still read through this?

**Pull request merged**